typescript 5.4.2 generates invalid types for `test`

**Describe the bug**
Latest `@oclif/test` uses typescript 5.4.2, which causes consumers to run into compilation errors:

```
node_modules/@oclif/test/lib/index.d.ts:3:28 - error TS2315: Type 'Fancy' is not generic.

  3 export declare const test: import("fancy-test").Fancy<import("fancy-test/lib/types").Context, {
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  4     skip: {
    ~~~~~~~~~~~
...
111     };
    ~~~~~~
112 }>;
    ~~
```

**To Reproduce**
Install latest `@oclif/test` into `@oclif/plugin-version` and run `yarn test`

**Expected behavior**
No compilation errors

**Screenshots**
![Screenshot 2024-03-11 at 10 16 45 AM](https://github.com/oclif/test/assets/10244328/4581f4d9-1403-497f-bb5c-1e00650d580d)

--- a/package.json
+++ b/package.json
@@ -5,9 +5,9 @@
   "author": "Salesforce",
   "bugs": "https://github.com/oclif/test/issues",
   "dependencies": {
-    "@oclif/core": "^3.19.2",
+    "@oclif/core": "^3.23.0",
     "chai": "^4.4.1",
-    "fancy-test": "^3.0.11"
+    "fancy-test": "^3.0.13"
   },
   "devDependencies": {
     "@commitlint/config-conventional": "^18.6.2",
@@ -20,15 +20,14 @@
     "eslint-config-oclif": "^5.1.1",
     "eslint-config-oclif-typescript": "^3.1.2",
     "eslint-config-prettier": "^9.1.0",
-    "globby": "^11.0.1",
-    "husky": "^8.0.3",
+    "husky": "^9.0.3",
     "lint-staged": "^15.2.2",
     "mocha": "^10",
     "nock": "^13.5.4",
     "prettier": "^3.2.5",
     "shx": "^0.3.3",
     "ts-node": "^10.9.2",
-    "typescript": "^5"
+    "typescript": "^5.4.2"
   },
   "engines": {
     "node": ">=18.0.0"
@@ -47,7 +46,7 @@
     "build": "shx rm -rf lib && tsc",
     "lint": "eslint . --ext .ts",
     "posttest": "yarn lint",
-    "prepare": "husky install",
+    "prepare": "husky && yarn build",
     "prepublishOnly": "yarn run build",
     "pretest": "yarn build --noEmit && tsc -p test --noEmit",
     "test": "mocha --forbid-only \"test/**/*.test.ts\""

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,4 +1,4 @@
-import {fancy} from 'fancy-test'
+import * as fancyTest from 'fancy-test'
 import {dirname} from 'node:path'
 
 import {command} from './command'
@@ -15,7 +15,7 @@ function traverseFilePathUntil(filename: string, predicate: (filename: string) =
   return current
 }
 
-// Update to path.dirname(url.fileURLToPath(import.meta.url)) whenever we update tsconfig target to ES2020
+// Update to path.dirname(url.fileURLToPath(import.meta.url)) whenever we migrate to ESM
 /* eslint-disable unicorn/prefer-module */
 loadConfig.root = traverseFilePathUntil(
   require.main?.path ?? module.path,
@@ -23,7 +23,8 @@ loadConfig.root = traverseFilePathUntil(
 )
 /* eslint-enable unicorn/prefer-module */
 
-export const test = fancy
+// Using a named export to import fancy causes this issue: https://github.com/oclif/test/issues/516
+export const test = fancyTest.fancy
   .register('loadConfig', loadConfig)
   .register('command', command)
   .register('exit', exit)

